src/storage/src/source/kafka.rs

```diff
@@ -820,6 +847,9 @@ impl SourceRender for KafkaSourceConnection {
                                 }
                             }
                         }
+                        if !partition_exhausted {
+                            notificator.notify_one();
```

Comment:
what's the purpose of doing this?

Reply:
this notificator is normally notified whenever there is data available by one of `librdkafka`'s background threads. `librdkafka` expects the application to consume all available messages whenever a notification is delivered. In other words, if you get notified and there are 1M messages ready to go but you only process say 10k of them it won't fire the notification again.

This change is introducing the possibility of waiting on the notificator without having processed all the available input. This is what `partition_exhausted` represents, the fact that there is more work to be done, and so if that is the case we notify the notificator ourselves so that the select at the bottom of this process loop gets immediately unblocked.